include/fs/littlefs.h

```diff
@@ -12,11 +12,22 @@
 #include <storage/flash_map.h>
 
 #include <lfs.h>
+#include <fs/fs.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+static inline bool littlefs_on_blkdev(struct fs_mount_t *mountp)
+{
+	if (IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV) &&
```

Comment:
shouldn't this just be:
` return mountp->flags & FS_MOUNT_FLAG_USE_DISK_ACCESS`

Reply:
The `IS_ENABLED()` will evaluate to literal 1 or 0, and if it evaluates to 0 we are hoping a compiler will simply optimize the check code to `return false`; this means that the flag will not be checked at all when the feature is not selected and code for that check should not be generated.

Reply:
@de-nordic, what about:
```
{
        if (!IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV)) {
                return false;
        }
        return mountp->flags & FS_MOUNT_FLAG_USE_DISK_ACCESS;
}
```

Reply:
@Laczen  - I think that @de-nordic  was for this solution. In that way, when `CONFIG_FS_LITTLEFS_BLK_DEV` is not defined (most cases for littlefs) the condition is false from the preprocessor stage of the compilation. Then the compiler just removes this code from the very outset.